keywords/TestServeriOS.py

```diff
@@ -0,0 +1,323 @@
+
+
+class TestServeriOS(TestServerBase):
+
+    def __init__(self, version_build, host, port, community_enabled=None, debug_mode=False):
+
+        super(TestServeriOS, self).__init__(version_build, host, port)
+        self.liteserv_admin_url = "http://{}:59850".format(self.host)
+        self.logfile_name = None
+        self.device_id = None
+        self.device = "iPhone-8-Plus"
+        self.debug_mode = False
+        if debug_mode:
+            self.debug_mode = True
+        if community_enabled:
+            self.app_dir = "CBLTestServer-iOS-community-{}".format(version_build)
+            self.package_name = "CBLTestServer-iOS-community-{}.zip".format(version_build)
+            self.app = "CBLTestServer-iOS"
+        else:
+            self.app_dir = "CBLTestServer-iOS-enterprise-{}".format(version_build)
+            self.package_name = "CBLTestServer-iOS-enterprise-{}.zip".format(version_build)
+            self.app = "CBLTestServer-iOS-EE"
+
+    def download(self, version_build=None):
+        """
+        1. Check to see if package is downloaded already. If so, return
+        2. Download the LiteServ package from latest builds to 'deps/binaries'
+        3. Unzip the packages and make the binary executable
+        """
+        if version_build is not None:
+            self.version_build = version_build
+        version, build = version_and_build(self.version_build)
+        app_name = "{}-{}.app".format(self.app, version_build)
+
+        expected_binary_path = "{}/{}/{}".format(BINARY_DIR, self.app_dir, app_name)
+        if os.path.isfile(expected_binary_path):
+            log_info("Package is already downloaded. Skipping.")
+            return
+
+        # Package not downloaded, proceed to download from latest builds
+        downloaded_package_zip_name = "{}/{}".format(BINARY_DIR, self.package_name)
+        url = "{}/couchbase-lite-ios/{}/{}/{}".format(LATEST_BUILDS, version, build, self.package_name)
+
+        log_info("Downloading {} -> {}/{}".format(url, BINARY_DIR, self.package_name))
+        resp = requests.get(url)
+        resp.raise_for_status()
+        with open("{}/{}".format(BINARY_DIR, self.package_name), "wb") as f:
+            f.write(resp.content)
+
+        extracted_directory_name = downloaded_package_zip_name.replace(".zip", "")
+        with ZipFile("{}".format(downloaded_package_zip_name)) as zip_f:
+            zip_f.extractall("{}".format(extracted_directory_name))
+
+        # Remove .zip
+        os.remove("{}".format(downloaded_package_zip_name))
+
+    def install_device(self):
+        """Installs / launches CBLTestServer on iOS device
+        Warning: Only works with a single device at the moment
+        """
+        if self.debug_mode:
+            self.app_name = "{}-{}-Device-debug.app".format(self.app, self.version_build)
+        else:
+            self.app_name = "{}-{}-Device.app".format(self.app, self.version_build)
+        # package_name = "CBLTestServer-iOS-Device.app"
```

Comment:
remove the commented code, if not needed.

Reply:
done